main.go

```diff
@@ -172,6 +176,24 @@ func main() {
 			"reconciliation for InferenceService, please provide --model-registry-inference-reconcile flag.")
 	}
 
+	kserveWithMeshEnabled, kserveWithMeshEnabledErr := utils.VerifyIfComponentIsEnabled(context.TODO(), mgr.GetClient(), utils.KserveAuthorinoComponent)
```

Comment:
why not use `context.Background()` ?

Reply:
No reason.
I can change it, if you think it is better.

Reply:
I went ahead, and changed it.

Reply:
I think it boils down to how we should deal with ctx propagation in general here. Setting it expliclty to `.TODO()` signals that we want to revise the whole approach and I think you can enable linter to catch those easily at the point the context propagation from reconcile down the chain is set. Not sure you are planning to do it, but I think that would be good to address this at some point.

About `context.Background()` - IIRC it is intended to be used at the highest level and is meant for background tasks. It essentially bypasses the benefits of context usage.

Reply:
Well... I don't plan to review later. And given this is a top-level call, I think `context.Background()` would be OK.